ui/components/component-library/button-base/button-base.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import Box from '../../ui/box';
+import { Icon, ICON_NAMES } from '../icon';
+import { Text } from '../text';
+
+import {
+  ALIGN_ITEMS,
+  BUTTON_SIZES,
+  JUSTIFY_CONTENT,
+  TEXT_COLORS,
+  DISPLAY,
+  TEXT,
+  SIZES,
+  FLEX_DIRECTION,
+} from '../../../helpers/constants/design-system';
+
+export const ButtonBase = ({
+  as = 'button',
+  block,
+  children,
+  className,
+  size = BUTTON_SIZES.MD,
+  icon,
+  iconPositionRight,
```

Comment:
What do you think about using this a an enum rather than a boolean and setting the default to `"left"`? Could maybe even create a const to store directions and store it in `button-base.constants.js` this also aligns to how mobile is doing it.

 ```suggestion
  iconPosition = 'left',
```

Reply:
Deciding to keep it as boolean after discussion